core/domain/story_domain.py

```diff
@@ -341,6 +348,11 @@ def validate(self):
             node.validate()
             if node.id == self.starting_node_id:
                 initial_node_is_present = True
+            # Checks whether the number in the id of any node is greater than
+            # the value of next_node_id.
+            if int(node.id.replace('node_', '')) >= self.next_node_id:
```

Comment:
For the first part of this clause, do something like node.get_number_from_node_id() instead. This behavior should be encapsulated in the Node object, you shouldn't be spreading details of how its id is generated outside where that information is needed.

Reply:
Done, added the function to class StoryNode.